### PR TITLE
chore(python): add Python 3.12 in test matrix

### PR DESCRIPTION
**Pull request by stegayet**
Please report changes to metadata to the upstream [libphonenumber](https://github.com/google/libphonenumber/)
project rather than here.

Do not send pull requests with changes in files under any of the following directories:

- `resources/`
- `python/phonenumbers/data/`
- `python/phonenumbers/carrierdata/`
- `python/phonenumbers/shortdata/`
- `python/phonenumbers/geodata/`
- `python/phonenumbers/tzdata/`

All of these directories hold code that is autogenerated from the metadata in the `resources/` directory, which is an
exact copy of the [equivalent directory](https://github.com/google/libphonenumber/tree/master/resources) in the
upstream project.

This PR adds Python 3.12 in test matrix.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,14 +9,15 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        python-version: [3.5, 3.6, 3.7, 3.8, 3.9, "3.10", 3.11, pypy2.7, pypy3.9]
+        python-version: [3.5, 3.6, 3.7, 3.8, 3.9, "3.10", 3.11, 3.12, pypy2.7, pypy3.9]
```
Comment (AA-Turner) [inline .github/workflows/ci.yml:12]: Should this be ``3.12`` or ``3.12-dev``? I've seen both used.

Comment (stegayet) [reply .github/workflows/ci.yml:12]: Using 3.12 and `allow-prereleases: true` will use the latest 3.12 build (3.12b4 at the time of writing this comment).

3.12-dev might also work.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -9,14 +9,15 @@ jobs:
...
 
     steps:
     - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435 # v4
+      uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
```
Comment (daviddrysdale) [inline .github/workflows/ci.yml]: Comment needs updating to indicate what version the updated hash refers to.

**Review by daviddrysdale — COMMENTED** [review]
Thanks!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (stegayet) [reply .github/workflows/ci.yml]: Done :smile:

**Review by daviddrysdale — APPROVED** [review]

**Pull request merged**